### PR TITLE
Fixed setting up mocks with the new firebase_core_platform_interface

### PR DESCRIPTION
**Pull request by OlegNovosad**
- [x] Fixed setting up mocks with the new firebase_core_platform_interface version
- [x] Updated dependencies
- [x] Fixed some lint issues

### CODE REVIEW

**Comment by OlegNovosad** [comment]
> Might as well move `flutter_lints` and `fluter_test` to `dev_dependencies` in `pubspec.yaml`, I suppose they belong there actually.

Done.

**Comment by OlegNovosad** [comment]
@sitatec I've also updated Github Actions build script with next improvements:

- Use the latest stable flutter
- Changed `runs-on` from `macos-latest` to `ubuntu-latest` to minimize actions time spent as macos minute is treated as 10 ubuntu minutes

**Comment by Zohenn** [comment]
Cool, that's what caused my PR to fail to build. I wanted to make similar changes, but now I'll wait for @sitatec to merge your PR.

File: test/mock_database_reference_test.dart [hunk 1]
```diff
@@ -19,7 +18,7 @@ void main() {
     });
     test("Should get reference using the deprecated reference() method", () {
       expect(
-        MockFirebaseDatabase.instance.reference().child("deprecated").path,
+        MockFirebaseDatabase.instance.ref().child("deprecated").path,
         MockDatabaseReference().child("deprecated").path,
       );
     });
```
Comment (sitatec) [inline test/mock_database_reference_test.dart:24]: We need to revert this change. This test is for the `reference` method not for the `ref` one, the `ref` method has it own tests.
Although the `reference` method is deprecated, it hasn't been removed yet.

**Review by sitatec — CHANGES_REQUESTED** [review]
Thanks for your contribution @OlegNovosad. We just need to revert one change before merging.

File: test/mock_database_reference_test.dart (hunk 1 above)
Comment (OlegNovosad) [reply test/mock_database_reference_test.dart]: Makes sense @sitatec ! Updated.

**Review by sitatec — APPROVED** [review]

**Pull request merged**